feature: Temp Url 기능 구현 진행

wannte opened the pull request:
#178 사전 동의 기능의 일부인 temp url을 잘라서 구현진행했습니다.

`/petitions/me` 부분에서 어떤 응답을 전달할지에 대해서 고민해봐야 합니다.

wannte committed:
feat: URL generator 구현

wannte committed:
feat: TempPetition 기능 구현

wannte committed:
feat: TempPetitionResponse 구현

koseyeon commented on src/test/java/com/gistpetition/api/utils/urlGenerator/RandomUrlGeneratorTest.java:
```diff
@@ -0,0 +1,22 @@
+    @Test
+    void generate() {
+        assertThat(tempUrlGenerator.generate(TEMP_URL_LENGTH)).hasSize(TEMP_URL_LENGTH);
+    }
```
randomURLGenerator인만큼 Generate를 두세번 진행해서 모두 다른게 나오는지 테스트하는건 어떨까요?

koseyeon commented on src/main/java/com/gistpetition/api/exception/petition/DuplicatedTempUrlException.java:
```diff
@@ -0,0 +1,12 @@
+import org.springframework.http.HttpStatus;
+
+public class DuplicatedTempUrlException extends PetitionException {
+    private static final String MESSAGE = "이미 청원에 링크가 존재합니다.";
```
클라이언트는 tempURL이 이미 존재한다는건 알게되었지만 이미 달린 tempURL의 주소를 알수 있는 방법이 없으니, 다시 청원을 등록해달라는 메시지를 주는건 어떨까요?

wannte committed:
fix: while 문 삭제

무한 루프가 돌 가능성이 있는 부분 삭제.

TempUrl이 한 개만 존재하는 알고리즘을 고민할 필요가 있음. 지금의 구현에서는 실패한 요청으로 나와서 새로 청원을 생성하도록 시나리오가 구성됨.

ChoiEungi commented on src/test/java/com/gistpetition/api/petition/application/TempPetitionServiceTest.java:
```diff
@@ -0,0 +1,56 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@SpringBootTest
+class TempPetitionServiceTest {
```
serviceTest를 상속받으면 좋을 것 같습니다

ChoiEungi committed:
refactor: serviceTest 상속으로 변경
